### PR TITLE
Switches to the use of enchant for looking up misspellings and adds misspellings feature for farsi. 

### PR DESCRIPTION
**Pull request by halfak**
This affords a substantial performance improvement.  

$ python demonstrate_spelling_speed.py 
Sending requests with default User-Agent.  Set 'user_agent' on api.Session to quiet this message.
Wordnet check took 8.90493106842041 seconds
Enchant check took 0.03188371658325195 seconds

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -101,11 +101,11 @@ system.  These are for ``scipy`` and ``numpy``.
 
 Linux Mint 17.1:
 
-1. ``sudo apt-get install g++ gfortran liblapack-dev python3-dev``
+1. ``sudo apt-get install g++ gfortran liblapack-dev python3-dev myspell-pt myspell-fa myspell-en``
```
Comment (he7d3r) [inline README.rst]: Isn't this supposed to be myspell-en-us? I tried with myspell-en and got this:
```
$sudo apt-get install myspell-en
Lendo listas de pacotes... Pronto
Construindo árvore de dependências       
Lendo informação de estado... Pronto
E: Impossível encontrar o pacote myspell-en
```

File: revscoring/languages/farsi.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+def is_misspelled_process():
+    def is_misspelled(word):
+        return not DICTIONARY.check(word)
+    return is_misspelled
+
+is_misspelled = LanguageUtility("is_misspelled", is_misspelled_process,
+                                depends_on=[])
+
+english = Language("revscoring.languages.english", [is_misspelled])
```
Comment (he7d3r) [inline revscoring/languages/farsi.py]: farsi?

**Comment by halfak** [comment]
@he7d3r can you take another look

**Comment by Ladsgroup** [comment]
It would be better if we use persian instead of farsi.

**Pull request merged**